alu_u32/src/bitwise/mod.rs

```diff
@@ -0,0 +1,257 @@
+extern crate alloc;
+
+use crate::pad_to_power_of_two;
+use alloc::vec;
+use alloc::vec::Vec;
+use columns::{Bitwise32Cols, COL_MAP, NUM_COLS};
+use core::mem::transmute;
+use valida_bus::{MachineWithGeneralBus, MachineWithRangeBus8};
+use valida_cpu::MachineWithCpuChip;
+use valida_machine::{instructions, Chip, Instruction, Interaction, Operands, Word};
+use valida_opcodes::{AND32, OR32, XOR32};
+use valida_range::MachineWithRangeChip;
+
+use p3_air::VirtualPairCol;
+use p3_field::PrimeField;
+use p3_matrix::dense::RowMajorMatrix;
+use p3_maybe_rayon::*;
+
+pub mod columns;
+pub mod stark;
+
+#[derive(Clone)]
+pub enum Operation {
+    And32(Word<u8>, Word<u8>, Word<u8>), // (dst, src1, src2)
+    Or32(Word<u8>, Word<u8>, Word<u8>),  // ''
+    Xor32(Word<u8>, Word<u8>, Word<u8>), // ''
+}
+
+#[derive(Default)]
+pub struct Bitwise32Chip {
+    pub operations: Vec<Operation>,
+}
+
+impl<F, M> Chip<M> for Bitwise32Chip
+where
+    F: PrimeField,
+    M: MachineWithGeneralBus<F = F> + MachineWithRangeBus8,
+{
+    fn generate_trace(&self, _machine: &M) -> RowMajorMatrix<M::F> {
+        let rows = self
+            .operations
+            .par_iter()
+            .map(|op| self.op_to_row(op))
+            .collect::<Vec<_>>();
+
+        let mut trace =
+            RowMajorMatrix::new(rows.into_iter().flatten().collect::<Vec<_>>(), NUM_COLS);
+
+        pad_to_power_of_two::<NUM_COLS, F>(&mut trace.values);
+
+        trace
+    }
+
+    fn global_sends(&self, machine: &M) -> Vec<Interaction<M::F>> {
```

Comment:
Not sure this is needed? I think we still have to `assert_bool` the (claimed) input bits, and if we do that, it seems like the output we compute must be in byte range.

Reply:
Yep, you're right. Had copied this file from one of the other ALU chips and forgot to remove.